src/App.js

```diff
@@ -1,5 +1,83 @@
+// modules
+const { Console } = require('@woowacourse/mission-utils');
+const InputError = require('./InputError');
+const ValidationError = require('./ValidationError');
+const Computer = require('./Computer');
+const Judge = require('./Judge');
+
+// constants
+const { GAME_SETTING, RESULT } = require('./utils/constants');
+const { MIN_NUMBER, MAX_NUMBER, NUMBER_COUNT } = GAME_SETTING;
+const { STRIKE } = RESULT;
+
 class App {
-  play() {}
+  play() {
+    Console.print('숫자 야구 게임을 시작합니다.');
+    this.playGame();
+  }
+
+  playGame() {
+    const computer = Computer.createUniqueNumbers(MIN_NUMBER, MAX_NUMBER, NUMBER_COUNT);
+    this.guess(computer);
+  }
+
+  guess(computer) {
+    Console.readLine('숫자를 입력해주세요 : ', (input) => {
+      try {
+        this.validateUserGuess(input);
+      } catch (err) {
+        if (err instanceof ValidationError) {
+          const message = `${MIN_NUMBER}부터 ${MAX_NUMBER}까지 서로 다른 ${NUMBER_COUNT}자리 숫자를 입력해주세요.`;
+          throw new InputError(message, err);
+        } else {
+          throw err;
+        }
```

Comment:
에러를 잡았다가 다시 던져주신 이유가 있을까요?:)

Reply:
에러 종류와 관계없이 동일한 방식으로 에러를 처리하는 것은 디버깅을 어렵게 만들기 때문에 좋지 않습니다.
생각해보니까 코드가 `this.validateUserGuess(input);` 한 줄이어서 다른 에러에 대해서 고민하지 않아도 되었을 것 같네요.

Reply:
아 이렇게하면 에러 추적에도 좋고 핸들러를 레이러링별로 레이어링해서 잡아낼 수 있겠네요!👍🏻 
코드를 보면서 많은 생각이드네요! 지금처럼 에러핸들링의 레이어를 유지하면서도 더 좋은 방법은 없을까 고민하게되네요!... 같이 고민해보면 좋을것 같아요! 
제가 생각해보건 트라이 캐치구문을 외부로 뺴내서 함수를 받아서 트라이캐치를 하도록 만들어 주면 어떨까요??!


```js

   excutewithErrorHandler(logicFunc, errorFunc){
     try{
         logicFunc();
     } catch(err){
         errorFunc(err);
         throw UnCatchedError();
     }
  }
```       


```suggestion
    excuteWithErrorHandler(
      ()=>{
        Cosnoe.readline('숫자를 입력해주세요: ', this.validateUserGuess);
        }, 
      (error)=> {
        if(error instanceof ValidateError){
          const message = `${MIN_NUMBER}부터 ${MAX_NUMBER}까지 서로 다른 ${NUMBER_COUNT}자리 숫자를 입력해주세요.`;
          throw new InputError(message, err);
         }
      throw err;
      });
      
```

Reply:
@nerdchanii
에러를 세분화한다면 말씀하신 것처럼 한번 더 래퍼로 감싸는 것이 로직과 분리되어 가독성이 좋을 것 같네요!
검사하는 로직이 `this.validateUserGuess`밖에 없는데 에러를 세분화하다보니 try catch문이 불필요하게 쓰인 것 같아요😭

Reply:
그래도 미래지향적인 코드베이스라고 생각한다면 좋을 것 같은걸요??!:-))👏🏻 아마 에러처리해야하는 레이어가 좀 더 존재했다면 더 빛을 발했을 것 같네요✨